nimpy.nim

```diff
@@ -1505,3 +1505,23 @@ proc `==`*(a, b: PyObject): bool =
     pyLib.PyObject_RichCompareBool(a.rawPyObj, b.rawPyObj, Py_EQ) == 1
   else:
     false
+
+macro asDict*(a: untyped): PPyObject =
```

Comment:
You can specify `untyped{nnkTableConstr}` (and don't need `expectKind`) so that it works better with overload resolution, and then rename the macro to `toDict` to be an overload with `toDict` down below.

Reply:
I had tried that, this cannot work:
(it's `untyped{nkTableConstr}`, not `untyped{nnkTableConstr}` btw)

because it runs into https://github.com/nim-lang/Nim/issues/17164, and indeed fails with this case:
```nim
let dict1 = toDict({"a": "foo", "b": 12})
```

```
Error: type mismatch:
 got '(string, int)' for '("Key", 12)' [tuple]
 but expected '(string, string)' [tuple]
```

Reply:
Ah that's unfortunate. But at least it should be possible to make a `toDict` macro that will do it's thing in case of `nnkTableConstr`, and forward to value-marshalling proc otherwise. I think it's beneficial to have one `toDict` name for this. `toDict` vs `asDict` is hard to remember.

Reply:
PTAL, see last commit